src/bootstrap/builder.rs

```diff
@@ -848,6 +848,7 @@ impl<'a> Builder<'a> {
         // things still build right, please do!
         match mode {
             Mode::Std => metadata.push_str("std"),
+            Mode::Codegen => metadata.push_str("codegen"),
```

Comment:
@alexcrichton I was getting "found two different crates with name `num_cpus` that are not distinguished by differing `-C metadata`. This will result in symbol conflicts between the two." and a similar error for `cc` without this change; does it seem reasonable to just add this? I didn't change Cargo.lock at all so my assumption is that how we're linking things together changed due to differing uses of syntax or something like that.

Reply:
A better fix might actually to be to just remove the `cc` and `num_cpus` dependencies from the `Cargo.toml` of librustc_codegen_llvm, that way it just pulls in the sysroot crates used by the compiler already

Reply:
Hm so this is a bit painful due to build scripts being built with the beta compiler always I believe which leads us to not being able to load cc and such from the sysroot there -- this addition works in practice so I'd like to land it for now.

Reply:
I don't think that's quite right, if you're getting conflicting crates then something in the `rustc_driver` dependency graph already links to the `cc` crate, and that's what's being loaded. If you delete this addition and delete the lines in `Cargo.toml` the build should work just fine.

Reply:
I guess maybe we can't ever conflict with build deps since those are always built separately? I'll try that.